### PR TITLE
PostgreSQL Related Query Path Fixes

### PR DESCRIPTION
**Pull request by zinic**
## Description

This changeset addresses a few tickets parted out into independent commits. This also contains some additional `prepare-for-codereview` changes layered on top.

## Motivation and Context

This PR addresses the following tickets:
* BED-4997 - Post-Processing Constraint Violations
* BED-5003 - CyPgSQL Directionality Reversed for Expansion Query

## How Has This Been Tested?

Several new test cases have been added to cover changes.

## Types of changes

- Chore (a change that does not modify the application functionality)
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

File: packages/go/analysis/post.go
Comment (mistahj67) [inline packages/go/analysis/post.go:148]: Nit, this isn't used outside this file, can we swap this to a private function?

File: packages/go/cypher/models/pgsql/translate/expansion.go
Comment (mistahj67) [inline packages/go/cypher/models/pgsql/translate/expansion.go:711]: Similar nit feedback here re: private vs public

File: packages/go/cypher/models/pgsql/translate/expansion.go
Comment (superlinkx) [inline packages/go/cypher/models/pgsql/translate/expansion.go:709]: It was mentioned during live review that this opens us up to a lot of nil pointers depending on use case. What mitigation strategies can we pursue until a more elegant structure presents itself? It seems like at a minimum, this type could be private as it's not used outside this package and given the volatile nature of it, probably shouldn't be used outside of this file

**Review by superlinkx — APPROVED** [review]
Slightly concerned by the nil pointers, but I don't think there's anything present as-is that would block this PR. Would like us to revisit that aspect in a follow up, so I'm approving as-is.

**Review by mistahj67 — COMMENTED** [review]
Really appreciate the cleanup 🫶 🧼

Why wasn't the edge constraint safety net not applied to the 
`ExecuteCommand` or `PostHybrid`? 

That also made me wonder why we didn't opt to move it one layer up in post processing as well, though I suspect that might be to minimize memory

File: packages/go/cypher/models/pgsql/translate/expansion.go
Comment (zinic) [reply packages/go/cypher/models/pgsql/translate/expansion.go]: Patching this up.

File: packages/go/cypher/models/pgsql/translate/expansion.go
Comment (zinic) [reply packages/go/cypher/models/pgsql/translate/expansion.go]: Patching this up.

File: packages/go/analysis/post.go
Comment (zinic) [reply packages/go/analysis/post.go:148]: I like this being a public contract for right now. It's the basic interaction of the datastructure and makes it more usable for little cost.

**Review by mistahj67 — APPROVED** [review]
Thanks for addressing nits! 🚀

**Review by superlinkx — APPROVED** [review]
Appreciate the due-diligence. This reads much safer and has my stamp of approval

**Pull request merged**
